internal/controller/postgrescluster/pgbackrest_test.go

```diff
@@ -2034,6 +2036,187 @@ func TestReconcilePostgresClusterDataSource(t *testing.T) {
 	}
 }
 
+func TestReconcileCloudBasedDataSource(t *testing.T) {
+	tEnv, tClient := setupKubernetes(t)
+	require.ParallelCapacity(t, 4)
```

Comment:
I decided to leave this at the same capacity of the ReconcilePostgresClusterDataSource test, since the ParallelCapacity currently blocks on anything greater than 0 (though technically this test doesn't create any pods, so maybe it could be tuned lower for when we do make ParallelCapacity respond more granularly?)